glyphs-reader/src/font.rs

```diff
@@ -1032,18 +1133,71 @@ fn user_to_design_from_axis_location(
     Some(axis_mappings)
 }
 
-fn extract_axis_mappings(from: &RawFont) -> BTreeMap<String, RawUserToDesignMapping> {
+fn add_mapping_if_new(
+    mappings: &mut RawUserToDesignMapping,
+    user: OrderedFloat<f32>,
+    design: OrderedFloat<f32>,
+) {
+    if mappings.iter().any(|(u, d)| *u == user || *d == design) {
+        return;
+    }
+    mappings.push((user, design));
+}
+
+/// * <https://github.com/googlefonts/glyphsLib/blob/6f243c1f732ea1092717918d0328f3b5303ffe56/Lib/glyphsLib/builder/axes.py#L128>
+/// * <https://github.com/googlefonts/glyphsLib/blob/6f243c1f732ea1092717918d0328f3b5303ffe56/Lib/glyphsLib/builder/axes.py#L353>
+fn add_mappings_from_instances(
+    instances: &[Instance],
+    mut mappings: BTreeMap<String, RawUserToDesignMapping>,
+) -> BTreeMap<String, RawUserToDesignMapping> {
+    for instance in instances
+        .iter()
+        .filter(|i| i.active && i.type_ == InstanceType::Single)
+    {
+        for (axis_name, inst_mapping) in instance.axis_mappings.iter() {
+            let mappings = mappings.entry(axis_name.clone()).or_default();
+            for (user, design) in inst_mapping.iter() {
+                add_mapping_if_new(mappings, *user, *design);
+            }
+        }
+    }
+    mappings
+}
+
+fn add_mappings_from_masters(
+    from: &RawFont,
+    mut mappings: BTreeMap<String, RawUserToDesignMapping>,
+) -> BTreeMap<String, RawUserToDesignMapping> {
+    for master in from.font_master.iter() {
+        let Some(axes) = from.axes.as_ref() else { continue; };
+        for (axis, value) in axes.iter().zip(master.axes_values.as_ref().unwrap()) {
+            let mappings = mappings.entry(axis.name.clone()).or_default();
```

Comment:
Maybe use different names here for clarity?

Reply:
Became nicer as a result of adding types